tests/test_pipeline_common.py

```diff
@@ -0,0 +1,507 @@
+                        f"{self.__class__.__name__}::test_pipeline_{task.replace('-', '_')} is skipped: test is "
+                        f"currently known to fail for: model `{model_architecture.__name__}` | tokenizer "
+                        f"`{tokenizer_name}` | processor `{processor_name}`."
+                    )
+                self.run_pipeline_test(task, repo_name, model_architecture, tokenizer_name, processor_name)
+
+    def run_pipeline_test(self, task, repo_name, model_architecture, tokenizer_name, processor_name):
+        """Run pipeline tests for a specific `task` with the give model class and tokenizer/processor class name
+
+        The model will be loaded from a model repository on the Hub.
+
+        Args:
+            task (`str`):
+                A task name. This should be a key in the mapping `pipeline_test_mapping`.
+            repo_name (`str`):
+                A model repository id on the Hub.
+            model_architecture (`type`):
+                A subclass of `PretrainedModel` or `PretrainedModel`.
+            tokenizer_name (`str`):
+                The name of a subclass of `PreTrainedTokenizerFast` or `PreTrainedTokenizer`.
+            processor_name (`str`):
+                The name of a subclass of `BaseImageProcessor` or `FeatureExtractionMixin`.
+        """
+        repo_id = f"hf-internal-testing/{repo_name}"
+
+        tokenizer = None
+        if tokenizer_name is not None:
+            tokenizer_class = getattr(transformers_module, tokenizer_name)
+            tokenizer = tokenizer_class.from_pretrained(repo_id)
+
+        processor = None
+        if processor_name is not None:
+            processor_class = getattr(transformers_module, processor_name)
+            # If the required packages (like `Pillow` or `torchaudio`) are not installed, this will fail.
+            try:
+                processor = processor_class.from_pretrained(repo_id)
+            except Exception:
+                self.skipTest(
+                    f"{self.__class__.__name__}::test_pipeline_{task.replace('-', '_')} is skipped: Could not load the "
+                    f"processor from `{repo_id}` with `{processor_name}`."
+                )
+
+        # TODO: Maybe not upload such problematic tiny models to Hub.
+        if tokenizer is None and processor is None:
+            self.skipTest(
+                f"{self.__class__.__name__}::test_pipeline_{task.replace('-', '_')} is skipped: Could not find or load "
+                f"any tokenizer / processor from `{repo_id}`."
+            )
+
+        # TODO: We should check if a model file is on the Hub repo. instead.
+        try:
+            model = model_architecture.from_pretrained(repo_id)
+        except Exception:
+            self.skipTest(
+                f"{self.__class__.__name__}::test_pipeline_{task.replace('-', '_')} is skipped: Could not find or load "
+                f"the model from `{repo_id}` with `{model_architecture}`."
+            )
```

Comment:
I would personally err towards raising an error for these models rather than skipping. Or would you rather not as it implies that there needs to be a hf-internal-testing model on the Hub for the tests to run?

Reply:
>  Or would you rather not as it implies that there needs to be a hf-internal-testing model on the Hub for the tests to run?

Yes. There are several cases where a model can't be created, some of them might be definite (i.e. nothing we can do), some of them I haven't dive into. So far, this information is not saved to a permeant place, so there is no way to check it and skip only for those one we definitely have no way to create. Error out them will fail the CI (even CircleCI). I will explain a bit more in our meeting.

Reply:
I don't get how this can happen without the user explicitly choosing to test (or not to test) a model with the pipeline mapping, now.

Reply:
As mentioned in [this earlier comment](https://github.com/huggingface/transformers/pull/21516#issuecomment-1437359847), I added these mapping using an automatic way, obtained from the current mappings in `xxxPipelineTest` classes, which produce complete lists of model classes to tests. I feel it's safer to keep what **should** be tested in the mapping, and skipping them if no tiny model on the Hub (i.e. to decouple these 2 things). See the last paragraph below.

There are some considerations regarding tiny model creation, the mapping I added in this PR, and the skip statements. I prefer not to change too many things in the PR, before a more detailed discussion, to avoid mistakes or more things to revert later. I would like to keep the logic as in the current `main`, and we can improve later 🙏.

One argument is: if we remove a model class from a mapping because a lack of tiny model on the Hub, we actually lose this information (lack of tiny model) on the test report, then it makes more unlikely checking and creating the tiny models and adding them back to the mapping.